pkg/solana/txm/pendingtx.go

```diff
@@ -547,6 +560,148 @@ func (c *pendingTxContext) TrimFinalizedErroredTxs() int {
 	return len(expiredIDs)
 }
 
+func (c *pendingTxContext) GetSignatureInfo(sig solana.Signature) (txInfo, error) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	info, exists := c.sigToTxInfo[sig]
+	if !exists {
+		return txInfo{}, ErrSigDoesNotExist
+	}
+	return info, nil
+}
+
+func (c *pendingTxContext) OnReorg(sig solana.Signature, id string) (pendingTx, error) {
+	err := c.withReadLock(func() error {
+		// Check if the transaction is still in a non finalized/errored state
+		var broadcastedExists, confirmedExists bool
+		_, broadcastedExists = c.broadcastedProcessedTxs[id]
+		_, confirmedExists = c.confirmedTxs[id]
+		if !broadcastedExists && !confirmedExists {
+			return ErrTransactionNotFound
+		}
+		return nil
+	})
+	if err != nil {
+		// If transaction or sig are not found, return
+		return pendingTx{}, err
+	}
+
+	var pTx pendingTx
+	// Acquire a write lock to perform the state reset if needed
+	_, err = c.withWriteLock(func() (string, error) {
+		// Retrieve tx again inside the write lock
+		tx, broadcastedProcessedExists := c.broadcastedProcessedTxs[id]
+		if broadcastedProcessedExists {
+			pTx = tx
+		}
+		tx, confirmedExists := c.confirmedTxs[id]
+		if confirmedExists {
+			pTx = tx
+		}
+
+		if !broadcastedProcessedExists && !confirmedExists {
+			// transaction does not exist in any non finalized/errored maps
+			return "", ErrTransactionNotFound
+		}
+
+		// Reset the transaction state to 'Broadcasted' upon detecting a reorg.
+		// Even if the transaction might have already progressed to 'Processed' before the reorg,
```

Comment:
nit: I think this comment is only relevant if the tx progressed after the re-org but our confirmation polling was too slow
```suggestion
		// Even if the transaction might have already progressed to 'Processed' after the reorg,
```

Reply:
If we make the change from my other comment, we probably wouldn't even have to wait till the next polling cycle to advance the state. We would just be marking the tx as `Processed` right after we're done handling the re-org. We can probably adjust this comment accordingly if you make that change